Web IDE fixes

uncomputable opened the pull request:
Some fixes that will make my life easier in the web IDE. Bump of MSRV to 1.63.0.

apoelstra commented on src/node/redeem.rs:
```diff
@@ -299,7 +299,9 @@ impl<J: Jet> RedeemNode<J> {
                 let target_ty = arrow.target.finalize()?;
-                self.bits.read_value(&target_ty).map_err(Error::from)
+                self.bits
+                    .decode_padded_value(&target_ty)
```
In 0719071b8f119db5345a62863a8a96ddf8cc594a:

The redeemnode encoding definitely does not use padded witness values. Does this really not break any tests?

uncomputable replied:
You are right. The witness values are encoded without padding. It's amazing that our unit tests don't fail.

I will introduce a value decoder with padding (for frames, etc.) and a decoder without padding (for witness data, etc.).

uncomputable replied:
The new methods force us to think about the presence / absence of padding, which is good.

apoelstra replied:
Thanks! And yes, absolutely. My original code had made the opposite mistake (assuming everything was using unpadded encoding).

uncomputable commented:
I moved the value decoding logic into associated methods of the `Value` struct. There are separate methods for compact and for padded encoding. A private trait is used for code de-duplication, which should be a zero-cost abstraction.

I also fixed some clippy lints.

apoelstra commented on src/value.rs:
```diff
@@ -388,6 +389,138 @@ impl<'a> Iterator for PaddedBitsIter<'a> {
+}
+
+struct CompactEncoding;
+struct PaddedEncoding;
```
In 34e1243f36022f88376fcaf810145d539a6aa669:

These should be empty enums. Otherwise they will have constructors.

uncomputable replied:
Thanks for pointing this out. I pushed a fix.

apoelstra reviewed (approved):
ACK 9ef06ef0280265432b7e2b1fd82492dd041c0aef successfully ran local tests
